internal/pkg/fieldtags/analyzer.go

```diff
@@ -147,32 +138,14 @@ func (sp *sourcePatterns) isSource(field *ast.Field) bool {
 	return false
 }
 
-// IsSource determines whether a FieldAddr is a source, that is whether it refers to a field previously identified as a source.
-func (t TaggedFields) IsSource(f *ssa.FieldAddr) bool {
-	n, ok := namedType(f)
-	if !ok {
-		return false
-	}
-	fName := utils.FieldName(f)
-	for _, tf := range t {
-		if tf.matches(n.String(), fName) {
-			return true
-		}
-	}
-	return false
-}
-
-func namedType(f *ssa.FieldAddr) (*types.Named, bool) {
-	structType := f.X.Type()
-	deref := utils.Dereference(structType)
-	named, ok := deref.(*types.Named)
-	return named, ok
-}
-
-func (t TaggedField) matches(typeName, fieldName string) bool {
-	return fmt.Sprintf("%s.%s", t.pkg, t.typ) == typeName && t.field == fieldName
+// IsSourceFieldAddr determines whether a ssa.FieldAddr is a source, that is whether it refers to a field previously identified as a source.
+func (t TaggedFields) IsSourceFieldAddr(fa *ssa.FieldAddr) bool {
+	// incantation plundered from the documentation for ssa.FieldAddr
```

Comment:
Carrying over a thread from a previous PR: would you like me to extract this to a function that makes all the intermediate checks to make sure that this can never panic?

(Since I took it straight from the documentation I would expect it to be fine.)

Reply:
I saw it's from the docstring on `ssa.FieldAddr`.  So I think the expression is fine, but I'd update `incantation plundered from` to explicitly point to `ssa.FieldAddr.Field docstring`

Reply:
SG. Done.